extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/context/TransactionContext.java

```diff
@@ -154,6 +169,12 @@ private Transaction getCurrentTransaction() {
         }
     }
 
+    void cleanupScope(Transaction transaction) {
+        synchronized (transactions) {
```

Comment:
after making the `transactions` Map a `ConcurrentMap`, the synchronized can be removed.